src/brc20-bridge/src/state/master_key.rs

```diff
@@ -0,0 +1,118 @@
+use bitcoin::bip32::ChainCode;
+use bitcoin::PublicKey;
+use ic_exports::ic_cdk::api::management_canister::ecdsa::{EcdsaCurve, EcdsaKeyId};
+use ic_stable_structures::stable_structures::Memory;
+use ic_stable_structures::{CellStructure as _, MemoryId, MemoryManager, StableCell, Storable};
+
+use crate::memory::MASTER_KEY_MEMORY_ID;
+
+pub struct MasterKeyStorage<M: Memory> {
+    master_key: StableCell<Option<MasterKey>, M>,
+}
+
+impl<M> MasterKeyStorage<M>
+where
+    M: Memory,
+{
+    /// Creates a new MasterKeyStorage.
+    pub fn new(memory: &dyn MemoryManager<M, MemoryId>) -> Self {
+        Self {
+            master_key: StableCell::new(memory.get(MASTER_KEY_MEMORY_ID), None)
+                .expect("stable memory master key initialization failed"),
+        }
+    }
+
+    /// Returns the master key if it exists.
+    pub fn get(&self) -> &Option<MasterKey> {
+        self.master_key.get()
+    }
+
+    /// Sets the master key.
+    pub fn set(&mut self, master_key: MasterKey) {
+        self.master_key
+            .set(Some(master_key))
+            .expect("failed to set master key");
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MasterKey {
+    pub public_key: PublicKey,
+    pub chain_code: ChainCode,
```

Comment:
What are the guarantees that the `PublicKey` and `ChainCode` struct do not change in the future? As these are external structs, we have no control on them, if they get changed our storage will be broken